app/Settings/GeneralSettings.php

```diff
@@ -42,11 +42,29 @@ public function settings_callback( $options ) {
 			}
 		}
 
+		if ($this->connected()) {
+			$show_free_shipping_banner = $options['show_free_shipping_banner'] == 1 ?: false;
+			TypesenseClient::get_instance()->site_info()->upsert( [ 
```

Comment:
Wouldn't it be better if we cache the site info into a variable and call upsert later?

Reply:
you can cache it by saving site_info into a variable.